src/frontend/admin/src/components/presentational/translatable-content/TranslatableForm.tsx

```diff
@@ -0,0 +1,137 @@
+import * as React from "react";
+import { PropsWithChildren, useEffect, useMemo, useRef, useState } from "react";
+import Box from "@mui/material/Box";
+import Tab from "@mui/material/Tab";
+import Tabs from "@mui/material/Tabs";
+import { TabContext } from "@mui/lab";
+import { useTheme } from "@mui/material/styles";
+import { TRANSLATE_CONTENT_LANGUAGE } from "@/utils/constants";
+import { LocalesEnum } from "@/types/i18n/LocalesEnum";
+import { getAcceptLanguage } from "@/services/http/HttpService";
+import { getLocaleFromDjangoLang } from "@/utils/lang";
+import { Maybe } from "@/types/utils";
+
+interface Props {
+  onSelectLang: (lang?: string) => void;
+  entitiesDeps?: any[];
+  resetForm?: () => void;
+}
+
+export function TranslatableForm({
+  entitiesDeps,
+  resetForm,
+  ...props
+}: PropsWithChildren<Props>) {
+  const langHasChanged = useRef<boolean>(false);
+  const formHasBeenReset = useRef<boolean>(false);
+  const [value, setValue] = useState(getAcceptLanguage());
+  const theme = useTheme();
+
+  const a11yProps = (index: number) => {
+    return {
+      id: `translate-tab-${index}`,
+      "aria-controls": `translate-tabpanel-${index}`,
+    };
+  };
+
+  const handleChange = (event: React.SyntheticEvent, newValue: string) => {
+    if (value === newValue) {
+      return;
+    }
+    setValue(newValue);
+    localStorage.setItem(TRANSLATE_CONTENT_LANGUAGE, newValue);
+    langHasChanged.current = true;
+    props.onSelectLang(newValue);
+  };
+
+  useEffect(() => {
+    /**
+     * If the language has changed, then we reset the form.
+     * If we reset the form, we set the formHasBeenReset property to true to mean that the form has been reset because
+     * the language has been changed.
+     */
+
+    if (langHasChanged.current) {
+      langHasChanged.current = false;
+      if (resetForm) {
+        resetForm();
+        formHasBeenReset.current = true;
+      }
+    }
+  }, entitiesDeps);
+
+  useEffect(() => {
+    localStorage.setItem(TRANSLATE_CONTENT_LANGUAGE, getLocaleFromDjangoLang());
+    const oldOnbeforeunload = window.onbeforeunload;
+    /*
+      The translation of content and the retrieval of an object according to a given language are done via the same
+      header on a GET / POST request. We play on the priorities in the "getAcceptLanguage" method of HttpService.
+      We add this event because if we are on a page with translatable content, we need to reset the
+      TRANSLATE_CONTENT_LANGUAGE key in localStorage when we leave or refresh the page so that the object is retrieved
+      in the current language and not in the current language forced. by the TranslatableContent component
+     */
+    window.onbeforeunload = () => {
+      localStorage.removeItem(TRANSLATE_CONTENT_LANGUAGE);
+    };
+    return () => {
+      window.onbeforeunload = oldOnbeforeunload;
+      localStorage.removeItem(TRANSLATE_CONTENT_LANGUAGE);
+      props.onSelectLang(getLocaleFromDjangoLang());
+    };
```

Comment:
Due to the nature of this event (triggered just before the page is leaving), we could assume to add a listener that can be ran once.

```suggestion
    window.addEventListener('beforeunload',() => {
      localStorage.removeItem(TRANSLATE_CONTENT_LANGUAGE);
    }, {once: true});

    return () => {
      localStorage.removeItem(TRANSLATE_CONTENT_LANGUAGE);
      props.onSelectLang(getLocaleFromDjangoLang());
    };
```